src/zkbob/ZkBobPoolETH.sol

```diff
@@ -0,0 +1,205 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.15;
+
+import "@openzeppelin/contracts/utils/Address.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@uniswap/v3-periphery/contracts/interfaces/ISwapRouter.sol";
+import "@uniswap/v3-periphery/contracts/interfaces/IPeripheryImmutableState.sol";
+import "@uniswap/v3-periphery/contracts/interfaces/external/IWETH9.sol";
+import "../interfaces/ITransferVerifier.sol";
+import "../interfaces/ITreeVerifier.sol";
+import "../interfaces/IBatchDepositVerifier.sol";
+import "../interfaces/IMintableERC20.sol";
+import "../interfaces/IOperatorManager.sol";
+import "../interfaces/IERC20Permit.sol";
+import "../interfaces/IPermit2.sol";
+import "../interfaces/IZkBobDirectDepositQueue.sol";
+import "../interfaces/IZkBobPool.sol";
+import "./utils/Parameters.sol";
+import "./utils/ZkBobAccounting.sol";
+import "../utils/Ownable.sol";
+import "../proxy/EIP1967Admin.sol";
+import "../utils/Sacrifice.sol";
+import "./ZkBobPool.sol";
+
+/**
+ * @title ZkBobETHPool
+ * Shielded transactions pool for native and wrappred native tokens.
+ */
+contract ZkBobPoolETH is ZkBobPool {
+    using SafeERC20 for IERC20;
+
+    IPermit2 public immutable permit2;
+
+    constructor(
+        uint256 __pool_id,
+        address _token,
+        ITransferVerifier _transfer_verifier,
+        ITreeVerifier _tree_verifier,
+        IBatchDepositVerifier _batch_deposit_verifier,
+        address _direct_deposit_queue,
+        address _permit2
+    )
+        ZkBobPool(__pool_id, _token, _transfer_verifier, _tree_verifier, _batch_deposit_verifier, _direct_deposit_queue)
+    {
+        require(Address.isContract(_permit2), "ZkBobPool: not a contract");
+        permit2 = IPermit2(_permit2);
+    }
+
+    // @inheritdoc ZkBobPool
+    function _withdrawNative(address user, uint256 tokenAmount) internal override returns (uint256 spentAmount) {
+        IWETH9(token).withdraw(tokenAmount);
+        if (!payable(user).send(tokenAmount)) {
+            new Sacrifice{value: tokenAmount}(user);
```

Comment:
We might want to start thinking about another approach, since self-destruct is gonna be deprecated this or next year - https://eips.ethereum.org/EIPS/eip-4760

So far, the only reasonable approach I can suggest is to allow native withdrawals only for EOA addresses.

@akolotov wdyt?

Reply:
I haven't seen such.
Just took a quick look at Aztec codebase, ETH from such failed withdrawals is kept on the contract balance, seems like with no way to recover it - https://github.com/AztecProtocol/aztec-connect/blob/a601e38f2866babae785fd9e0313ee02f208a754/contracts/src/core/processors/RollupProcessor.sol#L1262-L1269

Reply:
In case of WETH pool, acceptable solution might be to transfer WETH as a fallback in case of ETH transfer failure. I think we should go with that option for now.

As for the TokenSeller, I think we should wait until the concrete self-destruct deprecation EIP will be chosen and finalized.